source/common/stats/metric_impl.h

```diff
@@ -34,9 +39,27 @@ class MetricImpl : public virtual Metric {
     static const uint8_t Used = 0x1;
   };
 
+  void clear();
+
+private:
+  StatNameList stat_names_;
+};
+
+class NullMetricImpl : public MetricImpl {
+public:
+  explicit NullMetricImpl(SymbolTable& symbol_table)
+      : MetricImpl("", std::vector<Tag>(), symbol_table), symbol_table_(symbol_table),
+        stat_name_storage_("", symbol_table) {}
+  ~NullMetricImpl() { stat_name_storage_.free(symbol_table_); }
```

Comment:
Can there be some RAII wrapper that calls free automatically? Similar for any required `clear()` calls? Would like for this to be as fool-proof as possible. Same elsewhere.